example-app/src/main/java/com/adyen/checkout/example/ui/configuration/CheckoutConfigurationProvider.kt

```diff
@@ -49,65 +50,59 @@ internal class CheckoutConfigurationProvider @Inject constructor(
-        .addBacsDirectDebitConfiguration(getBacsConfiguration())
-        .addBcmcConfiguration(getBcmcConfiguration())
-        .addGooglePayConfiguration(getGooglePayConfiguration())
-        .add3ds2ActionConfiguration(get3DS2Configuration())
-        .addRedirectActionConfiguration(getRedirectConfiguration())
-        .addGiftCardConfiguration(getGiftCardConfiguration())
-        .setEnableRemovingStoredPaymentMethods(true)
-        .setAmount(amount)
-        .setAnalyticsConfiguration(getAnalyticsConfiguration())
-        .build()
-
-    fun getCardConfiguration(): CardConfiguration =
-        CardConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setShopperReference(keyValueStorage.getShopperReference())
-            .setAddressConfiguration(getAddressConfiguration())
-            .setInstallmentConfigurations(getInstallmentConfiguration())
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    private fun getCashAppPayConfiguration(): CashAppPayConfiguration =
-        CashAppPayConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setReturnUrl(CashAppPayComponent.getReturnUrl(context))
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    fun getBlikConfiguration(): BlikConfiguration =
-        BlikConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    fun getBacsConfiguration(): BacsDirectDebitConfiguration =
-        BacsDirectDebitConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    fun getInstantConfiguration(): InstantPaymentConfiguration =
-        InstantPaymentConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .build()
-
-    fun getGiftCardConfiguration(): GiftCardConfiguration =
-        GiftCardConfiguration.Builder(shopperLocale, environment, clientKey)
-            .setAmount(amount)
-            .setAnalyticsConfiguration(getAnalyticsConfiguration())
-            .setPinRequired(true)
-            .build()
+    val checkoutConfig: CheckoutConfiguration
+        get() = CheckoutConfiguration(
+            shopperLocale = shopperLocale,
+            environment = environment,
+            clientKey = clientKey,
+            amount = amount,
+            analyticsConfiguration = getAnalyticsConfiguration(),
+        ) {
+            // Drop-in
+            dropIn {
+                setEnableRemovingStoredPaymentMethods(true)
+            }
+
+            // Payment methods
+            bacsDirectDebit()
```

Comment:
The configurations with empty blocks could give the impression that we need to declare them even if we don't need to configure anything. WDYT about removing them?